tests/data_context/store/test_expectations_store.py

```diff
@@ -310,8 +310,262 @@ def test_get_key_in_non_cloud_mode(empty_data_context):
 @pytest.mark.unit
 def test_get_key_in_cloud_mode(empty_data_context_in_cloud_mode):
     cloud_data_context = empty_data_context_in_cloud_mode
-    project_manager.set_project(cloud_data_context)
     name = "test-name"
     suite = ExpectationSuite(expectation_suite_name=name)
     key = cloud_data_context.expectations_store.get_key(suite)
     assert isinstance(key, GXCloudIdentifier)
+
+
+@pytest.mark.cloud
+def test_add_expectation_success_cloud_backend(empty_cloud_data_context):
+    # Arrange
+    context = empty_cloud_data_context
+    store = context.expectations_store
+    suite_name = "test-suite"
+    suite = context.add_expectation_suite(suite_name)
+    key = store.get_key(suite)
+    expectation = ExpectColumnValuesToBeInSet(
+        column="a",
+        value_set=[1, 2, 3],
+        result_format="BASIC",
+    )
+
+    # Act
+    store.add_expectation(suite=suite, expectation=expectation)
+
+    # Assert
+    updated_suite_dict = store.get(key=key)
+    updated_suite = ExpectationSuite(**updated_suite_dict)
+    added_expectation = updated_suite.expectations[0]
+    assert UUID(added_expectation.id)
+    assert expectation.column == added_expectation.column
+    assert expectation.value_set == added_expectation.value_set
+    assert expectation.result_format == added_expectation.result_format
+
+
+@pytest.mark.filesystem
+def test_add_expectation_success_file_backend(empty_data_context):
+    # Arrange
+    context = empty_data_context
+    store = context.expectations_store
+    suite_name = "test-suite"
+    suite = context.add_expectation_suite(suite_name)
+    key = store.get_key(suite)
+    expectation = ExpectColumnValuesToBeInSet(
+        column="a",
+        value_set=[1, 2, 3],
+        result_format="BASIC",
+    )
+
+    # Act
+    store.add_expectation(suite=suite, expectation=expectation)
+
+    # Assert
+    updated_suite_dict = store.get(key=key)
+    updated_suite = ExpectationSuite(**updated_suite_dict)
+    added_expectation = updated_suite.expectations[0]
+    assert UUID(added_expectation.id)
+    assert expectation.column == added_expectation.column
+    assert expectation.value_set == added_expectation.value_set
+    assert expectation.result_format == added_expectation.result_format
+
+
+@pytest.mark.filesystem
+def test_add_expectation_disregards_provided_id(empty_data_context):
+    # Arrange
+    context = empty_data_context
+    store = context.expectations_store
+    suite_name = "test-suite"
+    suite = context.add_expectation_suite(suite_name)
+    key = store.get_key(suite)
+    provided_id = "e86bb8a8-b75f-4efb-a3bb-210b6440661e"
+    expectation = ExpectColumnValuesToBeInSet(
+        id=provided_id,
+        column="a",
+        value_set=[1, 2, 3],
+        result_format="BASIC",
+    )
+
+    # Act
+    store.add_expectation(suite=suite, expectation=expectation)
```

Comment:
What is expectation's id after this? Is it the new id or the old one?

Reply:
the store is unconcerned about IDs on add - it will always either set them to `None` (cloud mode) or overwrite them with a new `UUID`. The `ExpectationSuite` uses this method, and raises an error if an `Expectation` is passed to `add` with an ID.